httpcache.go

```diff
@@ -20,21 +20,45 @@ import (
 	"encoding/gob"
 	"fmt"
 	"io"
-	"log"
 	"net/http"
+	"strings"
 	"sync"
+	"time"
 
+	"github.com/buraksezer/olric"
+	"github.com/buraksezer/olric/config"
 	"github.com/caddyserver/caddy/v2"
 	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
-	"github.com/golang/groupcache"
-	"github.com/pomerium/autocache"
-	"github.com/pquerna/cachecontrol"
+	"github.com/pquerna/cachecontrol/cacheobject"
+	"go.uber.org/zap"
+)
+
+const userAgent = "Caddy"
+
+// db and dmap are global to preserve the cache between reloads and to share the same cache between all routes
+// These variables are destroyed and recreated if and only if there is a configuration change
+var (
+	db             *olric.Olric
+	dmap           *olric.DMap
+	previousConfig Config
 )
 
 func init() {
 	caddy.RegisterModule(Cache{})
 }
 
+type Config struct {
+	// Maximum size of the cache, in bytes. Default is 512 MB.
+	MaxSize    int64 `json:"max_size,omitempty"`
+	DefaultTTL int   `json:"default_ttl,omitempty"`
+	Olric      Olric `json:"olric,omitempty"`
```

Comment:
All exported config fields will need godoc comments, which are used as user-facing documentation on the website when registered, for example: https://caddyserver.com/docs/modules/http.handlers.webdav#field-list-header (you can even use markdown in them)

Not a blocker for now, but just FYI, it should be filled out.

Also, do you want to register this module on the Caddy website when it is ready for people to use? Either you or I can do it, I'm fine with either. Usually the primary maintainer should register it though.

Reply:
I'm fine with maintaining this module. I plan to add many more feature during the next weeks. I'll add the godoc comments!